vdom/src/com/vdom/core/CardPile.java

```diff
@@ -0,0 +1,133 @@
+package com.vdom.core;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import com.vdom.api.Card;
+
+public class CardPile  {
+	protected Card placeholderCard = null;
+	protected ArrayList<Card> cards;
+	protected ArrayList<Card> templateCards;
+
+	protected boolean allCardsVisible = true;
+	protected boolean isSupply = true;
+
+	protected boolean isBlackMarket = false;
+	protected boolean tradeRouteToken = false;
+
+	public CardPile(Card placeholder, Map<Card, Integer> cardList, boolean ordered, boolean allCardsVisible) {
```

Comment:
Don't use a Map here for the cardList - either use a List<Card> or use a List<CardMultiplicity> where the CardMultiplicity is a public static class inside CardPile containing a Card and an int.